langchain4j-core/src/main/java/dev/langchain4j/model/chat/ChatLanguageModel.java

```diff
@@ -6,20 +6,84 @@
 import dev.langchain4j.data.message.ChatMessage;
 import dev.langchain4j.data.message.UserMessage;
 import dev.langchain4j.model.chat.request.ChatRequest;
+import dev.langchain4j.model.chat.request.ResponseFormat;
+import dev.langchain4j.model.chat.request.ResponseFormatType;
+import dev.langchain4j.model.chat.request.ToolChoice;
 import dev.langchain4j.model.chat.response.ChatResponse;
 import dev.langchain4j.model.output.Response;
 
 import java.util.List;
 import java.util.Set;
 
+import static dev.langchain4j.internal.Utils.isNullOrEmpty;
 import static java.util.Arrays.asList;
-import static java.util.Collections.emptySet;
 
 /**
+ * TODO review all javadoc in this class
  * Represents a language model that has a chat interface.
+ *
+ * @see StreamingChatLanguageModel
  */
 public interface ChatLanguageModel {
 
+    /**
+     * TODO
+     * <p>
+     * A temporary default implementation of this method is necessary
+     * until all {@link ChatLanguageModel} implementations adopt it. It should be removed once that occurs.
+     *
+     * @param chatRequest
+     * @return
+     */
+    @Experimental
+    default ChatResponse chat(ChatRequest chatRequest) {
+
+        ResponseFormat responseFormat = chatRequest.responseFormat();
+        if (responseFormat != null && responseFormat.type() == ResponseFormatType.JSON) {
+            // TODO check supportedCapabilities() instead?
+            throw new UnsupportedOperationException("JSON response type is not supported by this model provider");
+        }
+
+        Response<AiMessage> response;
+        if (isNullOrEmpty(chatRequest.toolSpecifications())) {
+            response = generate(chatRequest.messages());
+        } else {
+            if (chatRequest.toolChoice() == ToolChoice.ANY) {
+                if (chatRequest.toolSpecifications().size() == 1) {
+                    response = generate(chatRequest.messages(), chatRequest.toolSpecifications().get(0));
+                } else {
+                    throw new UnsupportedOperationException(
+                            "ToolChoice.ANY is currently supported only when there is a single tool");
```

Comment:
I don't understand this, in the description you say

> ANY - LLM is forced to call one or more tools from the list of available ones.

So why would you narrow it to a single tool? My understanding is that ToolChoice.ANY should FORCE the LLM to use a tool, how would you achieve that?

Reply:
Also, the AUTO and ANY values don't sound very descriptive to me, or maybe I'm just having trouble understanding the meaning. Should it be AUTO and FORCE instead?

Reply:
Regarding naming: I simply use the most common names LLM providers have: https://docs.google.com/spreadsheets/d/1hOsbbAz9-qXCrnq_TXW-Q_AIAlssPLz9/edit?gid=532150960#gid=532150960

Reply:
> So why would you narrow it to a single tool? My understanding is that ToolChoice.ANY should FORCE the LLM to use a tool, how would you achieve that?

Current `generate()` API allows either `AUTO` or `ANY` with a single tool:
- `AUTO`: generate(List<ChatMessage> messages, List<ToolSpecification> toolSpecifications)
- `ANY` with a single tool: `generate(List<ChatMessage> messages, ToolSpecification toolSpecification)`

This method is a temporary solution for backwards compatibility, once all implementations implement chat() method. this limitation will go away.

Forcing LLM to call a tool is achieved by setting `toolChoice`/`toolMode` parameter in LLM provider API

Reply:
Ok that makes sense then, thanks.
Regarding NONE, do you plan to add that, or is it expected that the user will just not define any `toolSpecifications`?

Reply:
We can add `NONE`, but I have no idea why would anyone supply tools (that cost quite a lot of tokens) and then configure `NONE`. Mabe to make LLM aware of some tools, but restruct it? IDK, sounds very strange.

Reply:
This would be an opt-in validation (the user would have to set `ToolChoice.NONE`, it wouldn't be the default)

Reply:
But yeah feel free to ignore, maybe it's not generally a problem to make sure that we pass an empty list of tools

Reply:
Yes, but what if user wants to set `NONE` and still supply tools (this is a valid case for in LLM APIs)?

Reply:
Right, and only some providers support this parameter. I think then the best solution will be to just leave this as is for now... We can always rethink later. If the user doesn't want to use tools, they will have to make sure the tool specs are empty.

Reply:
I find `ANY` to be a confusing name. I'd prefer `REQUIRED` or maybe `MANDATORY`?

Reply:
Yeah, I'd go with REQUIRED

Reply:
Agreed, let's use `REQUIRED` then!